newIDE/app/src/MainFrame/EditorTabs/EditorTabsHandler.js

```diff
@@ -289,3 +289,24 @@ export const getEventsFunctionsExtensionEditor = (
 
   return null;
 };
+
+export const moveTabToPosition = (
+  editorTabsState: EditorTabsState,
+  fromIndex: number,
+  toIndex: number
+): EditorTabsState => {
+  if (toIndex < fromIndex) toIndex += 1;
+  const currentEditorTabs = [...getEditors(editorTabsState)];
+  const movingTab = currentEditorTabs[fromIndex];
+  currentEditorTabs.splice(fromIndex, 1);
+  currentEditorTabs.splice(toIndex, 0, movingTab);
+
+  let currentTabIndex: number = getCurrentTabIndex(editorTabsState);
```

Comment:
Is it required for you to cast this variable to a number? 
If so, then the function `getCurrentTabIndex` could be better typed to indicate it returns a number.